packages/shylu/shylu_dd/frosch/src/InterfaceSets/FROSch_DDInterface_def.hpp

```diff
@@ -547,59 +547,196 @@ namespace FROSch {
+//        
+//        Teuchos::ArrayView<const GO> indices2;
+//        Teuchos::ArrayView<const SC> values2;
+//        for (LO i=0; i<NumMyNodes_; i++) {
+//            commMat->getGlobalRowView(NodesMap_->getGlobalElement(i),indices2,values2);
+//            componentsSubdomains[i].resize(indices2.size());
+//            componentsSubdomainsUnique[i].resize(indices2.size());
+//            for (LO j=0; j<indices2.size(); j++) {
+//                componentsSubdomains[i][j] = indices2[j];
+//                componentsSubdomainsUnique[i][j] = indices2[j];
+//            }
+//            sortunique(componentsSubdomains[i]);
+//            sortunique(componentsSubdomainsUnique[i]);
+//        }
+//        
+//        sortunique(componentsSubdomainsUnique);
+//        
+//        return 0;
+//    }
+
     template <class SC,class LO,class GO,class NO>
     int DDInterface<SC,LO,GO,NO>::communicateLocalComponents(GOVecVecPtr &componentsSubdomains,
-                                                             GOVecVec &componentsSubdomainsUnique)
+                                                             GOVecVec &componentsSubdomainsUnique,
+                                                             UN priorDofsPerNode)
     {
-        Teuchos::RCP<Xpetra::Matrix<SC,LO,GO,NO> > commMat = Xpetra::MatrixFactory<SC,LO,GO,NO>::Build(NodesMap_,10);
-        Teuchos::RCP<Xpetra::Matrix<SC,LO,GO,NO> > commMatTmp = Xpetra::MatrixFactory<SC,LO,GO,NO>::Build(UniqueNodesMap_,10);
+        
+
+        if (UniqueNodesMap_->getMinAllGlobalIndex() > 0 || priorDofsPerNode > 0) {
+            GO minGID = UniqueNodesMap_->getMinAllGlobalIndex();
+            ConstGOVecView ElementList;
+            MapPtr tmpUniqueNodesMap;
+            {
+                GOVec indices(UniqueNodesMap_->getNodeNumElements());
+                ElementList = UniqueNodesMap_->getNodeElementList();
+                for (UN i=0; i<indices.size(); i++) {
+                    indices[i] = ElementList[i] - minGID;
+                }
+                tmpUniqueNodesMap = Xpetra::MapFactory<LO,GO,NO>::Build(UniqueNodesMap_->lib(),-1,indices(),0,UniqueNodesMap_->getComm());
+            }
+
+            MapPtr tmpNodesMap;
+            {
+                GOVec indices(NodesMap_->getNodeNumElements());
+                ElementList = NodesMap_->getNodeElementList();
+                for (UN i=0; i<indices.size(); i++) {
+                    indices[i] = ElementList[i] - minGID - (GO) priorDofsPerNode;
+                }
+                tmpNodesMap = Xpetra::MapFactory<LO,GO,NO>::Build(NodesMap_->lib(),-1,indices(),0,NodesMap_->getComm());
+            }
+
+            Teuchos::RCP<Xpetra::Matrix<SC,LO,GO,NO> > commMat = Xpetra::MatrixFactory<SC,LO,GO,NO>::Build(tmpNodesMap,10);
+            Teuchos::RCP<Xpetra::Matrix<SC,LO,GO,NO> > commMatTmp = Xpetra::MatrixFactory<SC,LO,GO,NO>::Build(tmpUniqueNodesMap,10);
 #ifdef Tpetra_issue_1752
```

Comment:
Do the `if` and the `else` block both work? If so, I think we should finally remove the `if` block.